### PR TITLE
Setup docker and make endpoints return data.

### PR DESCRIPTION
**Pull request by rendon**
I setup Docker as it is a fantastic technology for our application to run on the same environment in development and in production, this should help reduce inconsistencies, which usually happen because the environment used in development does not match the one used on production.

Also updated `/v1/results` and `/v1/standings` to return static JSON data, and added `/v1/problems/sets`. This static content will eventually be replaced with real data, we're returning fixed content so that the frontend can start obtaining data from the API.

### CODE REVIEW

File: controllers/problems_controller.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+type ProblemsController struct {
+}
+
+func (t *ProblemsController) SetIndex(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	data, err := problems.GetSets()
+	if err != nil {
+		log.Printf("Failed to retrieve results: %s", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, `{"status": 500, "message": "Failed to retrieve data"}`)
```
Comment (dventura11) [inline controllers/problems_controller.go]: Change 500 for http.StatusInternalServerError constant

File: controllers/problems_controller.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/algoristas/api/problems"
+)
+
+type ProblemsController struct {
```
Comment (dventura11) [inline controllers/problems_controller.go]: Add comments in Exported members

File: main.go [hunk 3]
```diff
@@ -13,5 +14,9 @@ func main() {
 	resultsController := controllers.NewResultsController()
 	http.HandleFunc("/v1/results", resultsController.Index)
 
+	problemsController := controllers.NewProblemsController()
+	http.HandleFunc("/v1/problems/sets", problemsController.SetIndex)
+
+	fmt.Println("Listening at :8080...")
```
Comment (dventura11) [inline main.go]: log instead of fmt

File: problems/problems.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package problems
+
+import (
+	"io/ioutil"
+	"os"
+)
+
+func GetSets() ([]byte, error) {
```
Comment (dventura11) [inline problems/problems.go]: I think that we should add interfaces to problems, results and standings to be possible mocking in unit test.

File: controllers/problems_controller.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
+package controllers
```
Comment (dventura11) [inline controllers/problems_controller.go]: Unit test please

**Comment by rendon** [comment]
@dventura11 I added some tests and some docs to the code. For the database I'm using the DAO pattern, I'm not 100% happy with the code though, I'll probably change a lot of things later.

**Pull request merged**